superduperdb/db/mongodb/cdc/handler.py

```diff
@@ -35,120 +34,54 @@ def __init__(self, db: DB, stop_event: threading.Event):
 
         threading.Thread.__init__(self, daemon=False)
 
-    def submit_task_workflow(
-        self, cdc_query: t.Optional[Serializable], ids: t.Sequence, task: str = "copy"
-    ) -> None:
-        """submit_task_workflow.
-        A fxn to build a taskflow and execute it with changed ids.
-        This also extends the task workflow graph with a node.
-        This node is responsible for applying a vector indexing listener,
-        and copying the vectors into a vector search database.
-
-        :param cdc_query: A query which will be used by `db._build_task_workflow` method
-        to extract the desired data.
-        :param ids: List of ids which were observed as changed document.
-        :param task: A task name to be executed on vector db.
-        """
-        if task == "delete":
-            task_workflow = TaskWorkflow(self.db)
-        else:
-            task_workflow = self.db._build_task_workflow(
-                cdc_query, ids=ids, verbose=False
-            )
-
-        task_workflow = self.create_vector_listener_task(
-            task_workflow, cdc_query=cdc_query, ids=ids, task=task
-        )
-        task_workflow.run_jobs()
-
-    def create_vector_listener_task(
-        self,
-        task_workflow: TaskWorkflow,
-        cdc_query: t.Optional[Serializable],
-        ids: t.Sequence[str],
-        task: str = 'copy',
-    ) -> TaskWorkflow:
-        """create_vector_listener_task.
-        A helper function to define a node in taskflow graph which is responsible for
-        executing the defined ``task`` on a vector db.
-
-        :param task_workflow: A DiGraph task flow which defines task on a di graph.
-        :param db: A superduperdb instance.
-        :param cdc_query: A basic find query to get cursor on collection.
-        :param ids: A list of ids observed during the change
-        :param task: A task name to be executed on vector db.
-        """
+    def run(self):
+        try:
+            for c in queue_chunker(CDC_QUEUE, self._stop_event):
+                _submit_task_workflow(self.db, Packet.collate(c))
+
+        except Exception as exc:
+            traceback.print_exc()
+            logging.info(f'Error while handling cdc batches :: reason {exc}')
+
+
+def _submit_task_workflow(db: DB, packet: Packet) -> None:
+    """
+    Build a taskflow and execute it with changed ids.
+
+    This also extends the task workflow graph with a node.
+    This node is responsible for applying a vector indexing listener,
+    and copying the vectors into a vector search database.
+    """
+    if packet.is_delete:
+        workflow = TaskWorkflow(db)
+    else:
+        workflow = db._build_task_workflow(packet.query, ids=packet.ids, verbose=False)
+
+    task = 'delete' if packet.is_delete else 'copy'
```

Comment:
may be DBEvent.delete DBEvent.copy?

Reply:
right, yes, doh!

Reply:
But wait, there is no DBEvent.copy!

I'm going to leave this the way it is, which exactly duplicates what is there, and then later see where these strings are actually going!